sylvia-derive/src/message.rs

```diff
@@ -518,54 +541,45 @@ impl<'a> MsgVariant<'a> {
         }
     }
 
-    fn extract_return_type(ret_type: &ReturnType) -> String {
-        let ty = if let ReturnType::Type(_, ty) = ret_type {
-            ty
-        } else {
+    fn extract_return_type(ret_type: &ReturnType) -> &PathSegment {
+        let ReturnType::Type(_, ty) = ret_type  else {
             unreachable!()
         };
-        let type_path = if let Type::Path(type_path) = ty.as_ref() {
-            type_path
-        } else {
+
+        let Type::Path(type_path) = ty.as_ref()  else {
             unreachable!()
         };
         let segments = &type_path.path.segments;
-        assert!(!segments.is_empty());
-        let args = if let PathArguments::AngleBracketed(arg) = &segments[0].arguments {
-            &arg.args
-        } else {
+        assert_eq!(segments.len(), 1);
+        let segment = &segments[0];
+
+        // In case of aliased result user need to define the return type by hand
+        if segment.ident != "Result" && segment.ident != "StdResult" {
+            panic!(
+                "Neither Result nor StdResult found in return type. \
+                    You might be using aliased return type. \
+                    Please use #[msg(return_type=<your_return_type>)]"
+            );
```

Comment:
Don't panic to error report. It works to some extend, but better use https://docs.rs/proc-macro-error/latest/proc_macro_error/macro.emit_error.html - it allows you to provide proper span, and in general improves how you can report an error.